tuftool/src/add_key_role.rs

```diff
@@ -66,8 +74,18 @@ impl AddKeyArgs {
     async fn add_key(&self, role: &str, mut editor: TargetsEditor) -> Result<()> {
         // create the keypairs to add
         let mut key_pairs = HashMap::new();
-        for source in &self.new_keys {
-            let key_source = parse_key_source(source)?;
+        let default_password = String::new();
+        let new_passwords = match &self.new_passwords {
+            Some(pws) => pws,
+            None => &vec![],
+        };
+
+        if new_passwords.len() > self.new_keys.len() {
+            panic!("More new passwords provided than new key sources");
```

Comment:
```suggestion
            error::MoreNewPasswordsSnafu.fail()?;
```

Would it be possible to replace any panics outside of tests with [snafu](https://docs.rs/snafu/latest/snafu/index.html) errors?

_errors.rs:_
```rust
    #[snafu(display("More new passwords provided than new key sources"))]
    MoreNewPasswords { backtrace: Backtrace },
```